### PR TITLE
Refactoring: Prioritization/sequence of deserialization

### PR DESCRIPTION
**Pull request by larshp**
#3536

This changes the determination of the deserialization sequence, so it can easily be extracted to the object handlers. All unit tests added in #5193 are passing.

Each object type declares the objects it requires, note how this can be very generic or very specific.

The object requirements are added to a graph, and removed from the graph by finding vertexes with no ingoing edges.

This might change some implicit sorting order, breaking some stuff, but 🤷‍♂️ we have to get this refactored.

### CODE REVIEW

File: src/objects/core/zcl_abapgit_file_deserialize.clas.locals_imp.abap [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        to-obj_name = ls_vertex-obj_name
+        TRANSPORTING NO FIELDS.
+      IF sy-subrc <> 0.
+        remove_vertex( lv_index ).
+        rs_item = ls_vertex.
+        RETURN.
+      ENDIF.
+    ENDLOOP.
+
+    READ TABLE mt_vertices INTO rs_item INDEX 1.
```
Comment (jfilak) [inline src/objects/core/zcl_abapgit_file_deserialize.clas.locals_imp.abap:81]: I would inform users that we have a cycle which cannot be resolved. Maybe a TODO comment here would be enough or  `IF ABAP_GIT_DEBUG == 'X'.  assert 1=0.  ENDIF.`

**Comment by mbtools** [comment]
We might want to use the graph class for a "deletion graph" later (refactor [`zcl_abapgit _dependencies`](https://github.com/abapGit/abapGit/blob/main/src/objects/core/zcl_abapgit_dependencies.clas.abap)). Let's make it global.

**Comment by larshp** [comment]
@jfilak warning is now added to log

@mbtools local graph class now extracted to new global class `zcl_abapgit_item_graph`

**Comment by mbtools** [comment]
The logic looks ok to me. It will have an impact on the order in which items are processed and performance. 

Order:

The graph is built with the complete object list. However, this happens before deserialize splits the list into three steps. While it works today, there will be a limitation as to what dependencies can be specified (for example, DDIC will always be before code). 

Performance:

With the proposed logic, the graph will be larger than necessary, since objects are added to the graph based on object type not based on the real dependencies. This might impact runtime for large projects. 

We should certainly follow through and move the logic to the object classes. However, to get the real dependencies will require processing the XML, which is not available from the current call point. We don't want to read the XML twice. So this needs some extra thought on how it should work in the future.

Overall, some concerns but I think it's safe to go ahead with this.

**Pull request merged**